angular/src/lib/modal/examples/modal-default.component.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, ViewChild } from '@angular/core';
+import { ModalComponent } from '@collab-ui/angular';
+
+@Component({
+  selector: 'example-modal-default',
+  template: `
+    <cui-modal-body [ngClass]="'custom-ng-class'">22222</cui-modal-body>
+    <a (click)='showModal()'>open</a>
+    <cui-modal #modal [whenHide]='onHide()' class='myClass' sizeType='small'>
+      <cui-modal-header headerLabel='Title' message='message'  ></cui-modal-header>
+      <cui-modal-body [ngClass]="'custom-ng-class'">Body</cui-modal-body>
+      <cui-modal-footer>
+        <a (click)='closeModal()'>close</a>
+      </cui-modal-footer>
+    </cui-modal>
+  `
+})
+export class ExampleModalDefaultComponent {
+
+  @ViewChild('modal') modal: ModalComponent;
+
+  constructor() {
```

Comment:
I think we should try to find a way to open/close the modal without a ViewChild ref @ArthusLiang. Otherwise its another thing for a user to add to make it work.

Reply:
> Can we see if we can make the modal open/close work without adding viewchild ref

There is a close button in modal-header. I do know if it is good to bind open feature by passing an element to modal. Do you have any other good suggestions?

Reply:
Talked about it with Arthus, this ViewChild Ref solution looks ok as the component isn't able to go off a variable change - true/false to open/close the modal.  The other option we talked about was using a service -- discussed how the service should be used internally on the component not an example. @pauljeter